corehq/apps/data_cleaning/tests/test_filter_outputs.py

```diff
@@ -0,0 +1,451 @@
+import uuid
+
+from django.test import TestCase
+
+from corehq.apps.data_cleaning.models import (
+    BulkEditFilter,
+    BulkEditSession,
+    DataType,
+    FilterMatchType,
+)
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.es.case_search import (
+    case_search_adapter,
+)
+from corehq.apps.es.client import manager
+from corehq.apps.es.tests.utils import es_test
+from corehq.apps.users.models import WebUser
+from corehq.form_processor.tests.utils import FormProcessorTestUtils
+from corehq.pillows.case_search import CaseSearchReindexerFactory
+from corehq.util.test_utils import create_and_save_a_case
+
+
+@es_test(requires=[case_search_adapter])
+class BulkEditFilterValueTests(TestCase):
+    domain = 'column-test-filters'
+    case_type = 'plant'
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.domain_obj = create_domain(cls.domain)
+        cls.addClassCleanup(cls.domain_obj.delete)
+
+        cls.web_user = WebUser.create(cls.domain, 'tester@datacleaning.org', 'testpwd', None, None)
+        cls.addClassCleanup(cls.web_user.delete, cls.domain, deleted_by=None)
+
+    def setUp(self):
+        super().setUp()
+        FormProcessorTestUtils.delete_all_cases()
+        self.session = BulkEditSession.objects.new_case_session(
+            self.web_user.get_django_user(),
+            self.domain,
+            self.case_type,
+        )
+
+    def _make_case(self, domain, case_properties, index=None):
+        case_properties = case_properties or {}
+        case_id = case_properties.pop('_id')
+        case_type = case_properties.pop('case_type', self.case_type)
+        case_name = f'case-name-{uuid.uuid4().hex}'
+        owner_id = case_properties.pop('owner_id', None)
+        case = create_and_save_a_case(
+            domain, case_id, case_name, case_properties, owner_id=owner_id, case_type=case_type, index=index
+        )
+        return case
+
+    def _bootstrap_cases_in_es_for_domain(self, domain, input_cases):
+        for case in input_cases:
+            index = case.pop('index', None)
+            self._make_case(domain, case, index=index)
+        CaseSearchReindexerFactory(domain=domain).build().reindex()
+        manager.index_refresh(case_search_adapter.index_name)
+
+    def _assert_filters_work_correctly(self, session, input_cases, expected_case_ids):
+        self._bootstrap_cases_in_es_for_domain(session.domain, input_cases)
+        query = session.get_queryset()
+        results = query.run().hits
+        actual_case_ids = [hit['_id'] for hit in results]
+        assert set(actual_case_ids) == set(expected_case_ids)
```

Comment:
Is there a specific reason for using `assert` over `self.assertEqual` here, or is either fine?

Reply:
@nospame `assert` is better. See [this comment thread](https://github.com/dimagi/commcare-hq/pull/36598#discussion_r2201445123). I think the web team is overdue some kind of statement on `#gtd-web-team`.